#! /bin/sh -e

# DP: Issue #4961: Inconsistent/wrong result of askyesno function in
# DP: tkMessageBox with Tcl8.5.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

Index: Lib/lib-tk/tkMessageBox.py
===================================================================
--- Lib/lib-tk/tkMessageBox.py	(revision 73494)
+++ Lib/lib-tk/tkMessageBox.py	(working copy)
@@ -70,11 +70,13 @@
     if title:   options["title"] = title
     if message: options["message"] = message
     res = Message(**options).show()
-    # In some Tcl installations, Tcl converts yes/no into a boolean
+    # In some Tcl installations, yes/no is converted into a boolean.
     if isinstance(res, bool):
-        if res: return YES
+        if res:
+            return YES
         return NO
-    return res
+    # In others we get a Tcl_Obj.
+    return str(res)
 
 def showinfo(title=None, message=None, **options):
     "Show an info message"
